app/controllers/courses/modern_languages_controller.rb

```diff
@@ -77,6 +74,18 @@ def selected_language_subjects
       end
     end
 
+    def selected_non_language_subjects
+      ids = params.fetch(:course, {})[:subjects_ids]
```

Comment:
```suggestion
      ids = params.dig(:course, :subjects_ids) || [] 
```

maybe

Reply:
6 and 2 threes?

Reply:
actually i prefer yours as this will always return a collection

Reply:
I've changed this now and removed the nil guard